libnopegl/src/node_eval.c

```diff
@@ -107,6 +127,38 @@ static const struct node_param eval_vec4_params[] = {
 
 NGLI_STATIC_ASSERT(variable_info_is_first, offsetof(struct eval_priv, var) == 0);
 
+static const char * const comp_selectors[] = {"0123", "rgba", "xyzw", "stpq"};
+
+static int register_component_names(struct hmap *vars, const char *base_name, int nb_components, float *ptr)
+{
+    int ret = 0;
+
+    ngli_assert(nb_components > 1);
+
+    const size_t len = strlen(base_name);
+    if (len >= SIZE_MAX - 3)
+        return NGL_ERROR_LIMIT_EXCEEDED;
+    const size_t size = len + 3;
+    char *name_comp = malloc(size);
+    if (!name_comp)
```

Comment:
Can't we use an hardcoded max len for the identifier like we do for glsl (and check snprintf return)?

Reply:
Thanks